lib/api_repositury/repository.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:github_repositories/api_repositury/data_models/repository_model.dart';
+import 'package:github_repositories/api_repositury/data_models/user_models.dart';
+import 'package:github_repositories/consts/strings.dart';
+import 'package:http/http.dart';
+import 'package:http/http.dart' as http;
+
+class Repository {
+  Future<UserModel> getUser(String name) async {
+
+    Map<String, String> headers = <String, String>{};
+    headers['Content-Type'] = 'application/json';
+    headers['Access-Control-Allow-Headers'] = 'Access-Control-Allow-Origin, Accept';
+
+    Uri url = Uri.parse(Strings.baseUrl + name);
+    Response response;
+    try {
+      response = await http.get(url, headers: headers);
+    } on Exception catch (e) {
+      return UserModel(error: Strings.unknownError);
+    } catch (e) {
+      return UserModel(error: Strings.unknownError);
+    }
+    if ((200 <= response.statusCode && response.statusCode < 300) || response.statusCode == 404) {
```

Comment:
I doubt that it should look like this. Your request should either throw an error or return success.
Like:
```
try {
// make request
// here is logic where your request was successful
} catch (e) {
// request failed
}
```